### PR TITLE
refactor: avoid string references for known modifiers

### PR DESCRIPTION
**Pull request by midgleyc**
1. Make "Evaluated Modifiers" a proper enum value instead of a special-cased string
2. Use functions taking Modifiers over functions taking Strings when looking up by enum.

### CODE REVIEW

File: src/net/sourceforge/kolmafia/KoLCharacter.java [hunk 1]
```diff
@@ -5377,7 +5380,7 @@ record StatExp(DoubleModifier exp, DoubleModifier pct) {}
           };
       Function<DoubleModifier, Double> calc =
           (DoubleModifier statPct) ->
-              (finalBaseExp + finalExp) * (1 + newModifiers.get(statPct) / 100.0f);
+              (finalBaseExp + finalExp) * (1 + newModifiers.getDouble(statPct) / 100.0f);
```
Comment (Veracity0) [inline src/net/sourceforge/kolmafia/KoLCharacter.java:5383]: I realize this is modified existing code, but, 100.0f? As in a single float?

File: src/net/sourceforge/kolmafia/Modifiers.java [hunk 2]
```diff
@@ -145,34 +146,34 @@ public Map<DerivedModifier, Integer> predict() {
...
-            + (int) Math.ceil(this.get(DoubleModifier.MUS_PCT) * mus / 100.0));
+            + (int) this.getDouble(DoubleModifier.MUS)
+            + (int) Math.ceil(this.getDouble(DoubleModifier.MUS_PCT) * mus / 100.0));
     rv.put(
         DerivedModifier.BUFFED_MYS,
         mys
-            + (int) this.get(DoubleModifier.MYS)
-            + (int) Math.ceil(this.get(DoubleModifier.MYS_PCT) * mys / 100.0));
+            + (int) this.getDouble(DoubleModifier.MYS)
+            + (int) Math.ceil(this.getDouble(DoubleModifier.MYS_PCT) * mys / 100.0));
```
Comment (Veracity0) [inline src/net/sourceforge/kolmafia/Modifiers.java:171]: Note that this one is not "f".

**Review by Veracity0 — CHANGES_REQUESTED** [review]
I saw a lot of those string names in your last PR but didn't comment because it was outside the scope of that PR.

File: src/net/sourceforge/kolmafia/KoLCharacter.java (hunk 1 above)
Comment (midgleyc) [reply src/net/sourceforge/kolmafia/KoLCharacter.java:5383]: To clarify: the change you want is to remove the `f`?

Comment (midgleyc) [reply src/net/sourceforge/kolmafia/KoLCharacter.java:5383]: There are more `f` in the stat lines ~5390 -- do you want them gone as well?

Comment (Veracity0) [reply src/net/sourceforge/kolmafia/KoLCharacter.java:5383]: Yes. when you are calculating a double, why force some of the factors to be simple floats?

**Review by Veracity0 — DISMISSED** [review]
Looks good. Thanks!

**Review by Veracity0 — APPROVED** [review]

**Pull request merged**